share/functions/__fish_print_hostnames.fish

```diff
@@ -5,10 +5,11 @@ function __fish_print_hostnames -d "Print a list of known hostnames"
 	if type -q getent
 		getent hosts 2>/dev/null | tr -s ' ' ' ' | cut -d ' ' -f 2- | tr ' ' '\n'
 	else if test -r /etc/hosts
-		tr -s ' \t' '  ' < /etc/hosts | sed 's/ *#.*//' | cut -s -d ' ' -f 2- | __fish_sgrep -o '[^ ]*'
+		tr -s ' \t' '  ' < /etc/hosts | sed 's/ *#.*//' | sed 's/^[0.]* .*$//' | cut -s -d ' ' -f 2- | __fish_sgrep -o '[^ ]*'
```

Comment:
I wasted some time trying to stringily it it - I was really wishing there was a negated behavior like `grep -v` available string. I failed to get it to work correctly and it was overcomplicated. I gave up before I got my crazy regex working just used sed. I guess I didn't use actual `grep -v` at that point because.. I'm not sure.